plenum/server/node.py

```diff
@@ -2528,35 +2550,36 @@ def processRequest(self, request: Request, frm: str):
 
     def is_query(self, txn_type) -> bool:
         # Does the transaction type correspond to a read?
-        handler = self.get_req_handler(txn_type=txn_type)
-        return handler and isinstance(handler,
-                                      LedgerRequestHandler
-                                      ) and handler.is_query(txn_type)
+        return self.read_manager.is_valid_type(txn_type)
+        # handler = self.get_req_handler(txn_type=txn_type)
+        # return handler and isinstance(handler,
+        #                               LedgerRequestHandler
+        #                               ) and handler.is_query(txn_type)
 
     def is_action(self, txn_type) -> bool:
-        return txn_type in self.actionReqHandler.operation_types
+        return self.action_manager.is_valid_type(txn_type)
+        # return txn_type in self.actionReqHandler.operation_types
 
     def can_write_txn(self, txn_type):
         return True
 
     def process_query(self, request: Request, frm: str):
         # Process a read request from client
-        handler = self.get_req_handler(txn_type=request.operation[TXN_TYPE])
         try:
-            handler.doStaticValidation(request)
+            self.read_manager.static_validation(request)
             self.send_ack_to_client((request.identifier, request.reqId), frm)
         except Exception as ex:
             self.send_nack_to_client((request.identifier, request.reqId),
                                      str(ex), frm)
-        result = handler.get_query_response(request)
+        result = self.read_manager.get_result(request)
         self.transmitToClient(Reply(result), frm)
 
     def process_action(self, request, frm):
         # Process an execute action request
         self.send_ack_to_client((request.identifier, request.reqId), frm)
         try:
-            self.actionReqHandler.validate(request)
-            result = self.actionReqHandler.apply(request)
+            self.action_manager.dynamic_validation(request)
```

Comment:
So, it looks like we have methods `static_validation` in ReadManager and `dynamic_validation` in ActionManager, while it's actually also static validation in ActionManager (since it's not ordered).